### DISCUSSION TITLE
Pulsar 2.10.1 will not run on m1 macos due to java.net.MalformedURLException

### DISCUSSION THREAD

**Discussion by LasseJacobs** [apache/pulsar#18433] [Q&A]
**Describe the bug**
When I run the docker command on the getting started page: https://pulsar.apache.org/docs/getting-started-docker/ on my M1 mac is fails during startup citing the following issue:
```
2022-07-24T10:40:57,598+0000 [main] ERROR org.apache.bookkeeper.common.component.AbstractLifecycleComponent - Failed to start Component: storage-service
java.util.ServiceConfigurationError: io.grpc.NameResolverProvider: Error accessing configuration file
	at java.util.ServiceLoader.fail(ServiceLoader.java:582) ~[?:?]
        ...
Caused by: java.net.MalformedURLException: no !/ found in url spec:file:/pulsar/lib/io.grpc-grpc-grpclb-1.45.1.jar!/META-INF/services/io.grpc.NameResolverProvider
	at java.net.JarURLConnection.parseSpecs(JarURLConnection.java:172) ~[?:?]
        ...
```

Full log is attached to this ticket.
[pulsar.log](https://github.com/apache/pulsar/files/9175699/pulsar.log)

After this error a number of other errors occur before the program exits, and the container stops running. I am using an ARM machine and this is an amd64 image. I ran the image on my linux server (with an amd64 processor) and the image runs fine. So I am assuming the issue is caused by the m1 processor/runtime environment. 

I'm hoping to find out if this is a solvable issue or not? In the past I have successfully been able to run amd64 images on my mac.

I tried searching through the logs for clues but I am not familiar with the project so I don't really know what to look for. There are a few warnings that mention: 
2022-07-24T10:40:56,373+0000 [main] WARN  org.apache.curator.retry.ExponentialBackoffRetry - maxRetries too large (2147483647). Pinning to 29

But I don't know how significant this is.

**To Reproduce**
1. Run `$ docker run -it -p 6650:6650  -p 8080:8080 --mount source=pulsardata,target=/pulsar/data --mount source=pulsarconf,target=/pulsar/conf apachepulsar/pulsar:2.10.1 bin/pulsar standalone`

**Expected behavior**
I hope to be able to run the image locally on my laptop.

**Desktop (please complete the following information):**
 - OS: macos BigSur, m1 CPU

**Comment by yaalsn** [comment]
@LasseJacobs How about running with command `docker run --platfrom linux/amd64 -it -p 6650:6650 -p 8080:8080 --mount source=pulsardata,target=/pulsar/data --mount source=pulsarconf,target=/pulsar/conf apachepulsar/pulsar:2.10.1 bin/pulsar standalone` ?

**Comment by tisonkun** [comment]
Work for me. So transferred to Q&A that it should be an env issue.
